### ISSUE TITLE
Person testing of feature 'report creating'

### ISSUE THREAD

**Issue by cefect** [NRCan/CanFlood#63]
- [x] install new QGIS version supported by CanFlood (3.22.8)
- [x] manually install new CanFlood dev version (need to create a zip of CanFlood folder. build dialog should read '1.2.0')
- [x] Review and edit the new toolset documentation ([5.3.4 Report Automator](https://canflood.readthedocs.io/en/dev/05_toolsets.html#report-automator))
- [x] Complete, review, and edit tutorial 2a ([section 6.2](https://canflood.readthedocs.io/en/dev/06_tutorials.html#tutorial-2a-risk-l2-with-simple-events))

Note I've created a[ new branch](https://github.com/NRCan/CanFlood/tree/63-person-testing-of-feature-report-creating) for this work.

Check the boxes here and create a pull request once this is finished.

**Comment by cefect** [comment]
Thanks for the review @blairscriven. As a result of your comments, I've made some fixes:

- fixed a bug that was preventing the curves from plotting
- uploaded [a pdf ](https://github.com/NRCan/CanFlood/blob/63-person-testing-of-feature-report-creating/tests2/data/test_t2_A_BuildDialog_0/CanFlood_report_res1_0805.pdf) showing the report results from tutorial 2a
- added some additional checks on the map generator (to make sure the selected layer is not empty)
- expanded the documentation for tutorial 2a report generator
- activated [an RTD build](https://canflood.readthedocs.io/en/63-person-testing-of-feature-report-creating/) for this branch
- fixed some tests

If you could re-run your review on these revisions that would be most helpful. 

Notes:

- if you're unable to use git, github should allow you to upload file revisions manually
- be sure to create a new zip/install to get the latest changes

**Comment by blairscriven** [comment]
Hey so I tested the recent fixes and I am getting python errors that I was not getting yesterday. It occurs when I press the 'Create Report' button on the Report tab and stops the processing at the "loaded 'r_ttl' w/ (7, 4)" step.

Python Errors:
…\canflood\results\dialog.py", line 861, in run_reporterwrkr.add_map(vlay=geo_vlay)
…\canflood\results\reporter.py", line 308, in add_map assert isinstance(vlay, QgsVectorLayer)

My QGIS info:
QGIS version - 3.22.8-Białowieża
Qt version - 5.15.3
Python version - 3.9.5
GDAL/OGR version - 3.5.0
PROJ version - 9.0.1
Pandas version - 1.1.3
Openpyxl version - 3.0.7
Scipy version - 1.6.2
xlrd version - 1.2.0
xlsxwriter version - 3.0.1
matplotlib version - 3.5.1

**Comment by blairscriven** [comment]
@cefect , I think you were right about the finv vectorlayer not being selected in the Join Geo combo-box. I made sure the right file was added and this time it worked - the summary map and the plot of vulnerability functions were generated in my report! Compared to the [PDF report](https://github.com/NRCan/CanFlood/blob/63-person-testing-of-feature-report-creating/tests2/data/test_t2_A_BuildDialog_0/CanFlood_report_res1_0805.pdf) you uploaded, mine was completely similar. 

So, I believe my test of the PDF reporter has been successful!
